CONTRIBUTING.md

```diff
@@ -129,6 +129,38 @@ Run `rush build -t <packagename>` to build a single project, and all local proje
 
 By default, Rush only displays things written to `STDERR`. If you want to see the full output, pass `--verbose` to any of the build commands. This is also true for script commands (see below).
 
+### Testing
+
+Once the project has successfully built, you can test any of our packages by executing `rushx test` from the root of the package that you want to test. This will automatically run the tests in both NodeJS and in a browser. To target these environments individually, you can run `rushx test:node` and `rushx test:browser`.
```

Comment:
Not all SDKs have the `build:test` or the `clean` command as part of the `test` command.
If we are suggesting this right after the `build`/`rebuild` step, we better fix the `test` command too. 

(At least log an issue)

Reply:
I'd like this PR to be the forcing factor to fix some general expectations out of our test set up.
@sadasant, Please review the scripts for all Track 2 packages and log an issue for when the `test` script does have the build step in it.

Reply:
> when the `test` script does have the build step in it

we need to work with EngSys team on this as they might have separated steps: build test, execute tests, etc. we don't want our build pipeline wasting time on duplicate work.

Reply:
I think I can do the work in coordination with Engineering system. What is not clear to me is what direction we should go:

- [ ] Build the tests inside of the script that runs the tests.
- [ ] Always build the tests in a separate command.

?

Reply:
I suggest leave the united `test` command for developers, and isolated commands for clean, build, build:test, run-test for the pipelines.

Reply:
I've moved the build and the clean steps to the same `test` and `test-ci` commands in the new test guidelines proposal: https://github.com/Azure/azure-sdk-for-js/pull/7784

It looks like this:

````
  "test": "npm run clean && npm run build:test && npm run test:node && npm run test:browser",
  "test-ci": "npm run clean && npm run build:test && npm run test-ci:node && npm run test-ci:browser",
```

Reply:
The EngSys team doesn't have to worry about this as they dont use the `test` script. They use `unit-test` for PR and `integration-test` for nightly

Reply:
The thing is Daniel is suggesting to remove `integration-test` and `unit-test` completely.

Reply:
not as part of this PR right?? This PR should reflect the state of things as is in the master branch. The test guidelines are still in progress and under discussion

Reply:
Not as part of this PR! This PR has no more mentions of unit or integration tests.